### ISSUE TITLE
Script not working because of 301 redirect

### ISSUE THREAD

**Issue by nyuszika7h** [blha303/IRCCloud-logger#8]
```pytb
(venv)melon@cadoth:~/IRCCloud-logger$ python logger.py
Thu Apr 21 18:33:20 2016 log started
Traceback (most recent call last):
  File "logger.py", line 404, in <module>
    for line in streamiter(tmpcookie):
  File "logger.py", line 54, in streamiter
    origin="https://www.irccloud.com")
  File "/home/melon/IRCCloud-logger/venv/src/websocket/websocket/__init__.py", line 214, in create_connection
    websock.connect(url, **options)
  File "/home/melon/IRCCloud-logger/venv/src/websocket/websocket/__init__.py", line 475, in connect
    self._handshake(hostname, port, resource, **options)
  File "/home/melon/IRCCloud-logger/venv/src/websocket/websocket/__init__.py", line 512, in _handshake
    raise WebSocketException("Handshake Status %d" % status)
websocket.WebSocketException: Handshake Status 301
```

**Comment by blha303** [comment]
I don't use irccloud anymore, so I can't support logger.

**Comment by Liamraystanley** [comment]
This looks like it may be an issue on IRCcloud's end. More specifically, according to their up-to-date documentation, `wss://api.irccloud.com/` is supposed to be used as the websocket host, however, this is returning a redirect. Even when pointing to `wss://www.irccloud.com/`, it seems as though their server is redirecting to https, rather than wss, and upgrading the connection.

Tl;dr Cloudflare looks like it may be breaking things. If they resolve that, I can update the repo with the proper endpoints.

Here is a bit of extra info from my testing:

##### With api.irccloud.com:

```
[hq] [~/test/websocket-client]# ./bin/wsdump.py wss://api.irccloud.com/ -vv
--- request header ---
GET / HTTP/1.1
Upgrade: websocket
Connection: Upgrade
Host: api.irccloud.com:443
Origin: http://api.irccloud.com:443
Sec-WebSocket-Key: <truncated>
Sec-WebSocket-Version: 13


-----------------------
--- response header ---
HTTP/1.1 301 Moved Permanently
Content-length: 0
Location: https://www.irccloud.com/
Connection: close
-----------------------
Handshake status 301
```

##### With www.irccloud.com:

```
[hq] [~/test/websocket-client]# ./bin/wsdump.py wss://www.irccloud.com/ -vv
--- request header ---
GET / HTTP/1.1
Upgrade: websocket
Connection: Upgrade
Host: www.irccloud.com:443
Origin: http://www.irccloud.com:443
Sec-WebSocket-Key: <truncated>
Sec-WebSocket-Version: 13


-----------------------
--- response header ---
HTTP/1.1 301 Moved Permanently
Server: cloudflare-nginx
Date: Wed, 27 Apr 2016 04:56:55 GMT
Content-Length: 0
Connection: keep-alive
Set-Cookie: <truncated>; expires=Thu, 27-Apr-17 04:56:55 GMT; path=/; domain=.irccloud.com; HttpOnly
Location: https://www.irccloud.com/
CF-RAY: <truncated>
-----------------------
Handshake status 301
```

I'll keep this updated if anything changes, unless someone gets to it first.

**Comment by Liamraystanley** [comment]
So, looks like IRCcloud's server will break if the websocket initiation handshake puts a port in the `Host` header. This is a common standard, and follows the http RFC, however supplying the port will cause the redirection to occur.

I've put in a modified websocket-client package which should be a temporary resolution if IRCcloud does in fact end up resolving the issue, though doubtful.
